sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineTranslator.java

```diff
@@ -1051,11 +1052,9 @@ public void translate(
     // IO Translation.
 
     registerTransformTranslator(
-        PubsubIO.Read.Bound.class, new PubsubIOTranslator.ReadTranslator());
+        PubsubUnboundedSource.class, new PubsubIOTranslator.ReadTranslator());
     registerTransformTranslator(
-        DataflowPipelineRunner.StreamingPubsubIOWrite.class,
-        new PubsubIOTranslator.WriteTranslator());
-
+        PubsubUnboundedSink.class, new PubsubIOTranslator.WriteTranslator());
```

Comment:
PubsubIOTranslator.ReadTranslator is a TransformTranslator<PubsubIO.Read.Bound<T>>.
The type doesn't match for the changes above.

Are you trying to override the new pubsubIO back to nativeIO for Dataflow?

If so, I think you want to remove the two registerTransformTranslator above, and add the overide in DataflowPipelineRunner.
Here is how I did it for BigQuery,Write:
https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/205/files